Lib/csv.py

```diff
@@ -447,3 +449,14 @@ def has_header(self, sample):
                     hasHeader -= 1
 
         return hasHeader > 0
+
+
+    def _guess_lineterminator(self, sample):
+        # Guess line terminator based on frequency
+        terminators = {le: sample.count(le) for le in ('\r\n', '\n', '\r')}
```

Comment:
@terryjreedy I changed detection method; most probably line terminators would be consistent so if _one_ of them is present, it would be **the one**